test/src/dftbp/api/mm/testers/test_intcharges.f90

```diff
@@ -0,0 +1,173 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2023  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+!> example code for changing internal charges inside calculation
+program test_intcharges
+  use, intrinsic :: iso_fortran_env, only : output_unit
+  use dftbplus
+  ! Only needed for the internal test system
+  use testhelpers, only : writeAutotestTag
+  implicit none
+
+  integer, parameter :: dp = kind(1.0d0)
+
+  integer, parameter :: nAtom = 3
+
+  ! H2O coordinates, atomic units
+  real(dp), parameter :: initialCoords(3, nAtom) = reshape([&
+      & 0.000000000000000E+00_dp, -0.188972598857892E+01_dp,  0.000000000000000E+00_dp,&
+      & 0.000000000000000E+00_dp,  0.000000000000000E+00_dp,  0.147977639152057E+01_dp,&
+      & 0.000000000000000E+00_dp,  0.000000000000000E+00_dp, -0.147977639152057E+01_dp], [3, nAtom])
+
+  ! Atomic number of each atom
+  integer, parameter :: atomTypes(nAtom) = [8, 1, 1]
+
+  ! list of atoms by atomic number
+  character(2), parameter :: atomTypeNames(10) = [character(2) ::&
+      & "H", "He", "Li", "Be", "B", "C", "N", "O", "F", "Ne"]
+
+  character(100), parameter :: slakoFiles(2, 2) = reshape([character(100) :: &
+      & "./O-O.skf", "./H-O.skf", "./O-H.skf", "./H-H.skf"], [2, 2])
+
+  character(1), parameter :: maxAngNames(4) = ["s", "p", "d", "f"]
+
+
+  type(TDftbPlus) :: dftbp
+  type(TDftbPlusInput) :: input
+
+  integer, allocatable :: species(:)
+  character(2), allocatable :: speciesNames(:)
+  real(dp) :: merminEnergy
+  real(dp) :: coords(3, nAtom), gradients(3, nAtom)
+  real(dp) :: atomCharges(nAtom), cm5Charges(nAtom), atomMasses(nAtom)
+  real(dp) :: q0(nAtom)
+  type(fnode), pointer :: pRoot, pGeo, pHam, pDftb, pMaxAng, pSlakos, pAnalysis, pCm5
+  type(fnode), pointer :: pParserOpts
+
+  character(:), allocatable :: DftbVersion
+  integer :: major, minor, patch
+  integer :: ii
+
+  !integer :: devNull
```

Comment:
Here are some commented-out lines in this file, which should be removed from my perspective.

Reply:
Done,  but were presumably originally intended as hints for implementers (they are preserved in test_extcharges.f90)